packages/mobile/src/account/UploadServiceDataWrapper.ts

```diff
@@ -0,0 +1,181 @@
+import { ensureLeading0x, eqAddress, Err, Ok, Result } from '@celo/base'
+import { Address, ContractKit } from '@celo/contractkit'
+import {
+  FetchError,
+  InvalidSignature,
+  OffchainDataWrapper,
+  OffchainErrors,
+} from '@celo/identity/lib/offchain-data-wrapper'
+import { buildEIP712TypedData, resolvePath, signBuffer } from '@celo/identity/lib/offchain/utils'
+import { toChecksumAddress } from '@celo/utils/lib/address'
+import { recoverEIP712TypedDataSigner } from '@celo/utils/lib/signatureUtils'
+import { SignedPostPolicyV4Output } from '@google-cloud/storage'
+// Use targetted import otherwise the RN FormData gets used which doesn't support Buffer related functionality
+import FormData from 'form-data/lib/form_data'
+import * as t from 'io-ts'
+import config from 'src/geth/networkConfig'
+
+const TAG = 'UploadServiceDataWrapper'
+
+const expirationTime = 5 * 1000 * 60 // 5 minutes
+
+// Hacky way to get Buffer from Blob
+// Note: this is gonna transfer the whole data over the RN bridge (as base64 encoded string)
+// and should be avoided for large files!
+function blobToBuffer(blob: Blob) {
+  const reader = new FileReader()
+  reader.readAsDataURL(blob)
+  return new Promise<Buffer>((resolve, reject) => {
+    reader.onerror = () => {
+      reject(reader.error)
+    }
+    reader.onloadend = () => {
+      const result = reader.result
+      if (typeof result !== 'string') {
+        reject(new Error('Unexpected result type'))
+      } else {
+        // Result looks like "data:application/octet-stream;base64,BLMHTkM..."
+        // Extract the base64 part
+        const base64 = result.substr(result.lastIndexOf(',') + 1)
+        resolve(Buffer.from(base64, 'base64'))
+      }
+    }
+  })
+}
+
+// This data wrapper is passed to the offchain accessors (ex. PrivateNameAccessor and PrivatePictureAccessor)
+// It provides the functionality for reading/writing data to the backend via Valora
+export default class UploadServiceDataWrapper implements OffchainDataWrapper {
+  signer: Address
+  self: Address
+
+  constructor(readonly kit: ContractKit, address: Address, dataEncryptionKey: Address) {
+    this.signer = dataEncryptionKey
+    this.self = address
+  }
+
+  sendFormData(url: string, formData: typeof FormData): Promise<string> {
+    return fetch(url, {
+      method: 'POST',
+      headers: formData.getHeaders(),
+      // Use getBuffer() which ends up transferring the body as base64 data over the RN bridge
+      // because RN doesn't support Buffer inside FormData
+      // See https://github.com/facebook/react-native/blob/b26a9549ce2dffd1d0073ae13502830459051c27/Libraries/Network/convertRequestBody.js#L34
+      // and https://github.com/facebook/react-native/blob/b26a9549ce2dffd1d0073ae13502830459051c27/Libraries/Network/FormData.js
+      body: formData.getBuffer(),
+    }).then((x) => {
+      if (!x.ok) {
+        Logger.error(TAG + '@writeDataTo', 'Error uploading ' + x.headers.get('location'))
```

Comment:
```suggestion
        Logger.error(TAG + '@sendFormData', 'Error uploading ' + x.headers.get('location'))
        throw new Error(`Error uploading CIP8 data, with status ${x.status}, text: ${await response.text()}`)
```

Reply:
If there is an error, it will be passed back to the original `write` call in `profileInfo.ts`, where an error will be thrown